packages/php/blueprint/src/Importers/ImportRunSql.php

```diff
@@ -49,4 +47,36 @@ public function process( $schema ): StepProcessorResult {
 	public function get_step_class(): string {
 		return RunSql::class;
 	}
+
+	/**
+	 * Check if the current user has the required capabilities for this step.
+	 *
+	 * @param object $schema The schema to process.
+	 *
+	 * @return bool True if the user has the required capabilities. False otherwise.
+	 */
+	public function check_step_capabilities( $schema ): bool {
+		// General check.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return false;
+		}
+
+		$sql           = $schema->sql->contents;
+		$sql_lowercase = strtolower( trim( $sql ) );
+
+		// Check for specific operations.
+		if ( strpos( $sql_lowercase, 'posts' ) !== false ) {
```

Comment:
Both this and the `users` one below are quite generic and could easily be included in another query context that doesn't relate to the posts or users table specifically.
Should we stick to `wp_posts`, `wp_postmeta` and same for the users with `wp_users` and `wp_usermeta`?
Otherwise I may even lean to changing this so that a user has all three for any sql query: `current_user_can( 'manage_options' ) && current_user_can( 'edit_posts' ) && current_user_can( 'edit_users' )`

Reply:
I'll change the logic to check all three for any SQL query.

Reply:
Updated in 76ca635b9501260f8aaedef6835a9c35bee784a3.